### PR TITLE
Various invariant fixes: 

### PR DESCRIPTION
**Pull request by grandizzy**
- NFT CT2, record bucket 7388 on settle when deposits in the tree is zero and entire collateral into lowest bucket
- bucket take properly clear out deposit, set fenwick like structure to same amount in invariant test

### CODE REVIEW

File: tests/forge/regression/ERC20Pool/RegressionTestReservesERC20Pool.t.sol [hunk 1]
```diff
@@ -523,6 +523,112 @@ contract RegressionTestReserveERC20Pool is ReserveERC20PoolInvariants {
 
         invariant_rate_R1_R2_R3_R4_R5_R6_R7_R8();
     }
+
+    function test_regression_bucket_take_F1_F2() external {
```
Comment (ith-harvey) [inline tests/forge/regression/ERC20Pool/RegressionTestReservesERC20Pool.t.sol:531]: 1 - 2 lines of comments at the top describing what this was breaking in the invariants would be useful.

File: tests/forge/regression/ERC721Pool/RegressionTestLiquidationERC721Pool.t.sol [hunk 2]
```diff
@@ -103,6 +103,22 @@ contract RegressionTestLiquidationERC721Pool is LiquidationERC721PoolInvariants
         invariant_bucket_B5_B6_B7();
     }
 
+    /*
+        Test was failing because not all buckets were checked for collateral.
```
Comment (ith-harvey) [inline tests/forge/regression/ERC721Pool/RegressionTestLiquidationERC721Pool.t.sol:107]: :pray:

File: tests/forge/regression/ERC20Pool/RegressionTestReservesERC20Pool.t.sol (hunk 1 above)
Comment (grandizzy) [reply tests/forge/regression/ERC20Pool/RegressionTestReservesERC20Pool.t.sol:531]: Not sure why fails 😀

Comment (grandizzy) [reply tests/forge/regression/ERC20Pool/RegressionTestReservesERC20Pool.t.sol:531]: added with https://github.com/ajna-finance/contracts/pull/795/commits/74f225bbec276835ac43463df0c375fb64634e66

File: tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol [hunk 3]
```diff
@@ -194,7 +194,7 @@ abstract contract UnboundedLiquidationPoolHandler is BaseHandler {
                 }
             }
 
-            _fenwickRemove(beforeBucketTakeVars.deposit - afterBucketTakeVars.deposit, bucketIndex_);
+            fenwickDeposits[bucketIndex_] = afterBucketTakeVars.deposit;
```
Comment (ith-harvey) [inline tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol:207]: Would suggest a comment here explaining:
"// assign value to fenwick tree to mitigate rounding error that could be created in a `_fenwickRemove` call"

**Review by mattcushman — APPROVED** [review]
lgtm

File: tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol (hunk 3 above)
Comment (grandizzy) [reply tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol:207]: but the error is not created in `_fenwickRemove` call if I correctly understand this

Comment (grandizzy) [reply tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol:207]: or actually rereading Matt's comment it is, added with https://github.com/ajna-finance/contracts/pull/795/commits/aecee1b23d65e44fca3e7af2bd0f0edf464bd8f2

**Review by ith-harvey — APPROVED** [review]

**Pull request merged**